dev/com.ibm.ws.microprofile.faulttolerance_git_fat_tck/fat/src/com/ibm/ws/microprofile/faulttolerance/tck/FaultToleranceGitTckLauncher.java

```diff
@@ -0,0 +1,160 @@
+                          "CWMFT0016E", // CWMFT5016E: The Bulkhead policy parameter value value -1....
+                          "CWMFT5017E", // CWMFT5017E: The Retry policy maximum duration of 500 Millis for the
+                          // org.eclipse.microprofile.fault.tolerance.tck.invalidParameters.RetryClientForValidationDelayDuration.serviceA
+                          // target is not valid as it must be greater than the delay duration of 1,000
+                          // Millis.
+                          "CWWKZ0002E");// CWWKZ0002E: An exception occurred while starting the application
+                                        // ftInvalidCB5. The exception message was:
+                                        // com.ibm.ws.container.service.state.StateChangeException:
+                                        // org.jboss.weld.exceptions.DefinitionException
+    }
+
+    /**
+     * Run the TCK (controlled by autoFVT/publish/tckRunner/tcl/tck-suite.html)
+     *
+     * @throws Exception
+     */
+    @Mode(TestMode.EXPERIMENTAL)
+    @Test
+    @AllowedFFDC // The tested exceptions cause FFDC so we have to allow for this.
+    public void runFreshMasterBranchTck() throws Exception {
+
+        File repoParent = new File(GIT_REPO_PARENT_DIR);
+        File repo = new File(repoParent, GIT_REPO_NAME);
+
+        MvnUtils.mvnCleanInstall(repo);
+
+        HashMap<String, String> addedProps = new HashMap<String, String>();
+
+        String apiVersion = MvnUtils.getApiSpecVersionAfterClone(repo);
+        System.out.println("Queried api.version is : " + apiVersion);
+        addedProps.put("api.version", apiVersion);
+
+        String tckVersion = MvnUtils.getTckVersionAfterClone(repo);
+        System.out.println("Queried tck.version is : " + tckVersion);
+        addedProps.put("tck.version", tckVersion);
+
+        // A command line -Dprop=value actually gets to here as a environment variable...
+        String implVersion = System.getenv("impl.version");
+        System.out.println("Passed in impl.version is : " + implVersion);
+        addedProps.put("impl.version", implVersion);
+
+        // We store a set of keys that we want the system to add "1.1" or "1.2" etc to
+        // depending on the pom.xml contents.
+        HashSet<String> versionedLibraries = new HashSet<>(Arrays.asList("com.ibm.websphere.org.eclipse.microprofile.faulttolerance"));
+        String backStopImpl = "1.0"; // Used if there is no impl matching the spec/pom.xml <version> AND impl.version is not set
+
+        MvnUtils.runTCKMvnCmdWithProps(server, "com.ibm.ws.microprofile.faulttolerance_fat_tck", this.getClass() + ":launchFaultToleranceTCK", addedProps,
+                                       versionedLibraries, backStopImpl);
+    }
+
+    @Mode(TestMode.LITE)
+    @Test
+    public void testThatDoesNothingAndCanAlwaysRunAndPass() {
+        String mode = System.getProperty("fat.test.mode");
```

Comment:
Can more naturally check the test mode by comparing `componenttest.custom.junit.runner.TestModeFilter.FRAMEWORK_TEST_MODE`, which is what the `@Mode` annotations key off of as well

Reply:
Moved to that.